library/webpack-plugins/react-universal-plugin.js

```diff
@@ -169,39 +169,24 @@ module.exports = function reactUniversalPlugin () {
             source,
             '',
             '// __webpack_js_chunk_information__',
-            `${this.requireFn}.jci = ${JSON.stringify({ universalChunkNames, manifest })};`
+            `${compilation.mainTemplate.requireFn}.jci = ${JSON.stringify({ universalChunkNames, manifest })};`
           ]
-          return this.asString(buf)
-        })
-        compilation.mainTemplate.plugin('global-hash', () => true)
-        normalModuleFactory.plugin('parser', (parser, parserOptions) => {
-          parser.plugin(`expression __webpack_js_chunk_information__`, ParserHelpers.toConstantDependency('__webpack_require__.jci'))
-          parser.plugin(`evaluate typeof __webpack_js_chunk_information__`, ParserHelpers.evaluateToString('array'))
+          return buf.join('\n')
         })
+        compilation.mainTemplate.hooks.globalHash.tap(p, () => true)
+        normalModuleFactory.hooks.parser.for('javascript/auto').tap(p, addParserHooks)
+        normalModuleFactory.hooks.parser.for('javascript/dynamic').tap(p, addParserHooks)
```

Comment:
no idea what's happening here with the auto/dynamic stuff?

Reply:
Check here for an explanation of these: https://medium.com/webpack/webpack-4-released-today-6cdb994702d4#b6a9

It essentially requires us to define in which types of files this hook should trigger.